hack/build.sh

```diff
@@ -9,8 +8,27 @@ BUILD_DATE=$(date --utc -Iseconds)
 
 mkdir -p bin
 
-LDFLAGS="-s -w "
-LDFLAGS+="-X github.com/medik8s/node-maintenance-operator/version.Version=${VERSION} "
-LDFLAGS+="-X github.com/medik8s/node-maintenance-operator/version.GitCommit=${COMMIT} "
-LDFLAGS+="-X github.com/medik8s/node-maintenance-operator/version.BuildDate=${BUILD_DATE} "
-GOFLAGS=-mod=vendor CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags="${LDFLAGS}" -o bin/manager main.go
+LDFLAGS_VALUE="-X github.com/medik8s/node-maintenance-operator/version.Version=${VERSION} "
+# additional values must be quoted when used with GOFLAGS!
+LDFLAGS_VALUE+="-X github.com/medik8s/node-maintenance-operator/version.GitCommit=${COMMIT} "
+LDFLAGS_VALUE+="-X github.com/medik8s/node-maintenance-operator/version.BuildDate=${BUILD_DATE} "
+# allow override for debugging flags
+LDFLAGS_DEBUG="${LDFLAGS_DEBUG:-" -s -w"}"
+LDFLAGS_VALUE+="${LDFLAGS_DEBUG}"
+# must be single quoted for use in GOFLAGS, and for more options see https://pkg.go.dev/cmd/link
+LDFLAGS="'-ldflags=${LDFLAGS_VALUE}'"
+
+# add ldflags to goflags
+export GOFLAGS+=" ${LDFLAGS}"
+echo "goflags: ${GOFLAGS}"
+
+# allow override and use zero by default- static linking
+export CGO_ENABLED=${CGO_ENABLED:-0}
+echo "cgo: ${CGO_ENABLED}"
+
+# export in case it was set
+if [ ! -z "${GOEXPERIMENT}" ]; then
```

Comment:
not needed, you can export an empty var without problems

Reply:
Not sure there is a value of trying to export nothing, but I don't mind of removing this check